otherlibs/systhreads/thread.mli

```diff
@@ -166,7 +168,7 @@ val default_uncaught_exception_handler : exn -> unit
 (** [Thread.default_uncaught_exception_handler] will print the thread's id,
     exception and backtrace (if available). *)
 
-val set_uncaught_exception_handler : (exn -> unit) -> unit
+val set_uncaught_exception_handler : (exn -> unit) @ portable -> unit
```

Comment:
[According to sherlocode](https://sherlocode.com/?q=set_uncaught_exception_handler) this seems rarely used.

Reply:
We found only a couple of uses of this function and one of them was incorrect.  So, going with the backwards incompatible change seems appropriate here.